inferno-framework/subscriptions-test-kit
# Fi-3237: Rubocop and Continuous Integration

# Summary

- Add rubocop
- Fix rubocop issues
  - This was a lot. Most of the changes here are to fix lines that were too long
  - In several spots, I had to refactor some things based on the CyclomaticComplexity rule:
    - `collect_requirements.rb`: This is the most change here. Rubocop was the impetus, but there is more to it. I realized a good chunk of the code was not super aligned with how the rest of the module works and it had some redundancy. I brought it more in line by following the declarative model the rest of the module uses rather than imperative. This makes it less brittle and hopefully clearer.
    - `requirements_coverage.rb`: Small refactor to address CyclomaticComplexity (and improve the code). I also added `rubocop:disable` in two places where I thought breaking up the methods would make them less clear.
    - `subscription_create_endpoint.rb`, `subscription_status_endpoint.rb`: Small refactors to address CyclomaticComplexity
- Add CI, which includes specs, rubocop, and requirements coverage

# Testing Guidance

- Run all the requirements rake tasks, confirm they still work as they should
- Run all the test suites, confirm nothing is broken
- Check CI output, confirm it looks and works as expected
  - See passing job [here](https://github.com/inferno-framework/subscriptions-test-kit/actions/runs/11276268270)
  - See (deliberately) failing requirements check [here](https://github.com/inferno-framework/subscriptions-test-kit/actions/runs/11276325274)
